libass/ass_cpu.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2020 rcombs <rcombs@rcombs.me>
+ *
+ * This file is part of libass.
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef LIBASS_CPU_H
+#define LIBASS_CPU_H
+
+typedef enum ASS_CPUFlags {
+    ASS_CPU_FLAG_NONE = 0x00,
+#define ASS_CPU_FLAG_NONE          0x00
+#define ASS_CPU_FLAG_X86           0x01
+#define ASS_CPU_FLAG_X86_SSE2      0x02
+#define ASS_CPU_FLAG_X86_AVX       0x04
+#define ASS_CPU_FLAG_X86_AVX2      0x08
+#define ASS_CPU_FLAG_X86_AVX512ICL 0x10
+    ASS_CPU_FLAG_MAX  = (ASS_CPU_FLAG_X86_AVX512ICL << 1) - 1,
+} ASS_CPUFlags;
+
+void ass_set_cpu_flags_mask(ASS_CPUFlags mask);
+
+ASS_CPUFlags ass_get_cpu_flags(void);
```

Comment:
I'm personally prefer not to use an enum name for values outside of an enum scope, but feel free to ignore me here.